### PR TITLE
Re-add removed Packaging.props lines related to IntelliSense packages

### PR DESCRIPTION
**Pull request by carlossanlop**
Back in Preview 6, the build process was properly placing the nuget intellisense xml files next to the built dlls, but [a recent change](https://github.com/dotnet/corefx/commit/c4c1985531cc1f25c805bbc4801b85c98e0dfee0#diff-ba98fbb097dbdca16daf37fed8c21357) removed the lines in Packaging.props necessary for this task.

### CODE REVIEW

**Comment by carlossanlop** [comment]
@joperezr and @ahsonkhan helped find out the root cause of the mismatch. We tested this change in my local machine and it worked. They can explain any additional details if necessary.
@ViktorHofer please take a look at this change.

File: eng/Packaging.props [hunk 1]
```diff
@@ -12,6 +12,10 @@
     <PackagePlatform Condition="'$(PackagePlatform)' == 'amd64'">x64</PackagePlatform>
     <NativePackagePath>$(MSBuildThisFileDirectory)src/Native/pkg</NativePackagePath>
 
+    <XmlDocPackage>Microsoft.Private.Intellisense</XmlDocPackage>
```
Comment (ericstj) [inline eng/Packaging.props]: Should this instead be in Version.props?  We don't typically put versions in random places in the repo.  Optimistically we'll one day ingest this through maestro as well.

**Comment by ericstj** [comment]
Maybe we should add validation to packaging that every ref has an XML file next to it?

File: eng/Packaging.props (hunk 1 above)
Comment (ericstj) [reply eng/Packaging.props]: Actually I think the only thing you need is XmlDocFileRoot.  See https://github.com/dotnet/arcade/blob/247c0290c554a68dd3e560dd88fb0da26353d9d7/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/PackageLibs.targets#L54


That can be defined using the properties that Viktor moved 
https://github.com/dotnet/corefx/commit/c4c1985531cc1f25c805bbc4801b85c98e0dfee0#diff-4e8b29963372e731e4662efaf22b414dR12

Alternatively we could change the packaging targets to use the dirs that @ViktorHofer’s change copied to.

Comment (ViktorHofer) [reply eng/Packaging.props]: I missed that `XmlDocFileRoot` is used in `PackageLibs.targets`, we should a comment to point that out.

**Comment by ViktorHofer** [comment]
@carlossanlop I hope you don't mind that I pushed into your branch. I fixed my mistake and centralized the intellisense package id.

**Comment by ViktorHofer** [comment]
We probably want to service that into 3.0?

**Comment by ericstj** [comment]
Changes made in Master right now will make it into Preview8, but I agree we should port this one for Preview7.  Can you validate that it fixes the regression than submit a PR for a port?  /cc @wtgodbe

**Comment by carlossanlop** [comment]
> @carlossanlop I hope you don't mind that I pushed into your branch. I fixed my mistake and centralized the intellisense package id.

Thank you. No, I didn't mind. It's been merged already :)

File: eng/Packaging.props [hunk 2]
```diff
@@ -12,6 +12,9 @@
     <PackagePlatform Condition="'$(PackagePlatform)' == 'amd64'">x64</PackagePlatform>
     <NativePackagePath>$(MSBuildThisFileDirectory)src/Native/pkg</NativePackagePath>
 
+    <!-- Used by PackageLibs.targets -->
+    <XmlDocFileRoot>$(NuGetPackageRoot)$(MicrosoftPrivateIntellisensePackageId)/$(MicrosoftPrivateIntellisensePackageVersion)/xmldocs/netcoreapp</XmlDocFileRoot>
```
Comment (ahsonkhan) [inline eng/Packaging.props]: Just confirming, do we no longer need `XmlDocDir` (from https://github.com/dotnet/corefx/pull/38151/files#diff-ba98fbb097dbdca16daf37fed8c21357L18)?

**Comment by ahsonkhan** [comment]
Have we verified that the xml IntelliSense files now do show up?

**Comment by wtgodbe** [comment]
@ahsonkhan yes, @ericstj and I verified locally that the most recent official build did have the .xml files where expected.

File: eng/Packaging.props (hunk 2 above)
Comment (ericstj) [reply eng/Packaging.props]: XmlDocDir is still defined, just in a different place: https://github.com/dotnet/corefx/pull/38151/files#diff-8b8f08ffbf7b863fb3700c1718eeb4cbR20

**Pull request merged**